File: iOS/TaskBridge/Utils/NavigationUtils.swift
import TaskBridgeCore

public extension NavigationRepository {
    func isCurrentRoot(tab: AppTab) async throws -> Bool {
        let currentTab = try await fetchCurrentTab()
        guard currentTab == tab else { return false }
        let activePath = try await fetchActivePath()
        guard let current = activePath?.current else { return false }

        switch tab {
        case .tasks:
            return current is NavigationDestinationTasksRoot
        case .templates:
            return current is NavigationDestinationTemplatesRoot
        case .reminders:
            return current is NavigationDestinationRemindersRoot
        default:
            return false
        }
    }
}

File: Android/src/androidMain/kotlin/com/taskbridge/android/utils/NavigationUtils.kt
package com.taskbridge.android.utils

import com.taskbridge.android.repository.NavigationRepository
import com.taskbridge.core.models.navigation.AppTab
import com.taskbridge.core.models.navigation.NavigationDestination

public suspend fun NavigationRepository.isCurrentRoot(tab: AppTab): Boolean {
    if (fetchCurrentTab() != tab) return false
    val current = fetchActivePath()?.current ?: return false
    return when (tab) {
        AppTab.TASKS -> current is NavigationDestination.TasksRoot
        AppTab.REMINDERS -> current is NavigationDestination.RemindersRoot
        AppTab.TEMPLATES -> current is NavigationDestination.TemplatesRoot
    }
}
